src/impl/android/external_commit_helper.cpp

```diff
@@ -89,13 +89,13 @@ ExternalCommitHelper::ExternalCommitHelper(RealmCoordinator& parent)
     int ret = mkfifo(path.c_str(), 0600);
     if (ret == -1) {
         int err = errno;
-        if (err == ENOTSUP) {
+        if (err == ENOTSUP || err == EACCES) {
             // Filesystem doesn't support named pipes, so try putting it in tmp instead
             // Hash collisions are okay here because they just result in doing
             // extra work, as opposed to correctness problems
             std::ostringstream ss;
 
-            std::string tmp_dir(getenv("TMPDIR"));
+            std::string tmp_dir = parent.get_temp_dir();
```

Comment:
There doesn't seem to be an error check if this exists? It looks like the `temp_dir` is entirely optional, in which case this is unsafe and should throw an exception i not set?